samples/nrf9160/azure_fota/src/main.c

```diff
@@ -244,8 +248,25 @@ void main(void)
 	modem_configure();
 	k_sem_take(&network_connected_sem, K_FOREVER);
 
-	static struct azure_iot_hub_config config = {
+#if IS_ENABLED(CONFIG_DEVICE_ID_USE_HW_ID)
+	char device_id[HW_ID_LEN];
+
+	err = hw_id_get(device_id, ARRAY_SIZE(device_id));
+	if (err) {
+		LOG_ERR("Failed to retrieve device ID");
+		return;
+	}
+#endif
+
+	struct azure_iot_hub_config config = {
```

Comment:
A comment stating here that `CONFIG_AZURE_IOT_HUB_DEVICE_ID` will be used if `.device_id.size` is 0 could be helpful for some users.